Tests/PostgreSQLTests/PostgreSQLConnectionTests.swift

```diff
@@ -342,6 +342,25 @@ class PostgreSQLConnectionTests: XCTestCase {
         _ = try categories.wait()
     }
 
+    func testNotifyAndListen() throws {
+        let notifyConn = try PostgreSQLConnection.makeTest()
+        let listenConn = try PostgreSQLConnection.makeTest()
+        let channelName = "Foo"
+        let messageText = "Bar"
+        var messageReceived = false
+        
+        try listenConn.listen(channelName) { text in
+            messageReceived = text == messageText
+        }.catch({ err in XCTFail("error \(err)") })
+
+        try notifyConn.notify(channelName, message: messageText).wait()
+
+        sleep(1) // Wait for any delay in the message being received
```

Comment:
How about creating a test expectation and waiting on that instead? Should be faster and more reliable than sleeping. See e.g. https://github.com/grpc/grpc-swift/blob/d247c61b4de064b121b40829df62c69849cafd6b/Tests/SwiftGRPCTests/EchoTests.swift#L72 for an example that works under Linux.

Reply:
is this part of the XCTestSuite? if so cool I am such a newb I had no idea there were these great tools for testing asynchronously I am going to look into that now.

Reply:
thanks that is such a great thing to know its going to make this Redis and other Nio based interfaces so much easier to work with

Reply:
acctually this is a bit harder then it should be mostly because depending on the implementation and configuration you may or may not get a  channel id in the package back from the postgresql connection. this of course could be gotten around, I welcome a PR

Reply:
thats just not true. I can show you screen shots of the failure where postgres only sends one null terminated string back from the server my prefered implementation would be,

```
struct PostgreSQLNotificationResponse: Decodable {
    /// The message coming from PSQL
    let channelId: String
    let message: String
    init(from decoder: Decoder) throws {
        let container = try decoder.singleValueContainer()
        _ = try container.decode(Int32.self) // message length
        _ = try container.decode(Int32.self) // process id of message
        channelId = try container.decode(String.self)
        message = try container.decode(String.self)
    }
}

```

but when I run that against a psql 9 server I get the following error !
```PostgreSQLConnectionTests.swift:361: error: -[PostgreSQLTests.PostgreSQLConnectionTests testNotifyAndListen] : failed - error ⚠️ [BitsError.nullTerminatedString: This was not available in the buffer]
```

I dont get that error against other psql servers(that are 10). But I do this relatively terrible coalesce thing so depending on the message format it decodes
```
struct PostgreSQLNotificationResponse: Decodable {
    /// The message coming from PSQL
    let message: String
    init(from decoder: Decoder) throws {
        let container = try decoder.singleValueContainer()
        _ = try container.decode(Int32.self) // message length
        _ = try container.decode(Int32.self) // process id of message
        let channelId = try container.decode(String.self)
        let message = try? container.decode(String.self)
        self.message = message ?? channelId
    }
}
```

Reply:
Sorry, what error do you get? That code block shows up as empty to me.

Also, if Postgres randomly only returns one string when the documentation says otherwise, I think that might need more investigation or explanation. I read the documentation that the message might be omitted if the caller of "notify" didn't provide one, but the channel name should be there.

Reply:
the code is pretty simplisitic I don't see how notify could work without a channel, I see where there is a a potential case insensitivity issue.... but  not what you are talking about